### PR TITLE
Source derivation extension

### PR DESCRIPTION
**Pull request by dthaler**
Initial strawman for a source derivation extension.

### CODE REVIEW

File: yaml/_SOUR [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "https://gedcom.io/terms/v7/OBJE": "{0:M}"
+  "https://gedcom.io/terms/v7/PAGE": "{0:1}"
+  "https://gedcom.io/terms/v7/QUAY": "{0:1}"
+  "https://gedcom.io/terms/v7/SNOTE": "{0:M}"
+  "https://gedcom.io/terms/v7/SOUR-DATA": "{0:1}"
+  "https://gedcom.io/terms/v7/SOUR-EVEN": "{0:1}"
+
+superstructures:
+  "https://gedcom.io/terms/v7/PAGE": "{0:1}"
+...
```
Comment (dthaler) [inline yaml/_SOUR:33]: also add `subsumes`?

**Comment by Norwegian-Sardines** [comment]
Is this "pull" actually implementing the Source within a Source using _SOUR?

**Comment by dthaler** [comment]
> Is this "pull" actually implementing the Source within a Source using _SOUR?

It's not a source _within_ a source per se, it's the source of a source.  For example, if you have a book that contains a transcription of something, and cites what it's transcribed from, then this is how the book source references the original.

**Comment by Norwegian-Sardines** [comment]
>It's not a source within a source per se, it's the source of a source. For example, if you have a book that contains a transcription of something, and cites what it's transcribed from, then this is how the book source references the original.

Then I would continue to disagree with adding this to the specification.

1) If the genealogist adding this "source of a source" (known as a "Secondary Source") to the GEDCOM and never reviewed the secondary source then they could be perpetuating a bad transcription.
2) if they did review/see the secondary source then they should create a separate citation for that document.

People just don't know or understand sourcing.  If (Option 1) You should always try to read and cite the original work (the primary source). If it is not possible to do this, you have to cite the original as contained in the secondary source. For example: Johnson 1930 as cited in James 1975).

This would mean that the Source_Record should contain all of the necessary information about the source and pointers to the secondary source that which you are citing.  Because the GEDCOM currently indicates that source used is the Source_Record, meaning:

1) The Fact indicates the Source we found FACT.SOUR is (James 1975)
2) The Source_Record for James 1975 should cite the information about Johnson 1930

Therefore, one option is that the Source_Record needs to have its own Citation_Structure that points to another Source_Record for Johnson 1930 including the citation wording found in James 1975.  This would represent the true path of discovery AND fit a potential model provided in a expanded Source_Record or template extension discussed in my thoughts previously.

Another alternative would be the option of creating a "Citation_Record" that would contain all pertinent information for a full citation and reducing the need for or deprecating the use of a Citation_Structure and the Source_Record. 

All of these alternatives were initially discussed in[ Issue #14](https://github.com/dthaler/gedcom-citations/issues/14) and I've added additional comments!

**Pull request merged**